### PR TITLE
[RESTEASY-2328] Use CopyOnWriteMap to prevent map copies with each We…

### PR DESCRIPTION
**Pull request by anilgursel**
…bTarget instance

### CODE REVIEW

**Comment by anilgursel** [comment]
Only `JDK: openjdk8 Java SERVER_VERSION=16.0.0.Final` failed. 

Screen Shot 2019-08-21 at 9 17 51 AM


The failure seems to be unrelated to this PR:

```
[ERROR] Failed to execute goal org.codehaus.mojo:buildnumber-maven-plugin:1.4:create (get-scm-revision) on project resteasy-dependencies: Execution get-scm-revision of goal org.codehaus.mojo:buildnumber-maven-plugin:1.4:create failed: Plugin org.codehaus.mojo:buildnumber-maven-plugin:1.4 or one of its dependencies could not be resolved: Could not transfer artifact backport-util-concurrent:backport-util-concurrent:jar:3.1 from/to google-maven-central (https://maven-central.storage-download.googleapis.com/repos/central/data/): GET request of: backport-util-concurrent/backport-util-concurrent/3.1/backport-util-concurrent-3.1.jar from google-maven-central failed: Connection reset -> [Help 1]
```

File: resteasy-core/src/main/java/org/jboss/resteasy/core/providerfactory/ResteasyProviderFactoryImpl.java [hunk 1]
```diff
@@ -1476,7 +1462,7 @@ public T getContext(Class type)
    @Override
    public Map<String, Object> getProperties()
    {
-      return Collections.unmodifiableMap(properties);
+      return properties;
```
Comment (asoldano) [inline resteasy-core/src/main/java/org/jboss/resteasy/core/providerfactory/ResteasyProviderFactoryImpl.java]: Is this really needed? I mean I believe the idea here was that getProperties returns an unmodifiable map, while getMutableProperies returns the actual mutable map. Is there a problem with this?

Comment (asoldano) [reply resteasy-core/src/main/java/org/jboss/resteasy/core/providerfactory/ResteasyProviderFactoryImpl.java]: Thanks, I'll be merging soon!

**Comment by asoldano** [comment]
Merged, thanks @anilgursel !

**Pull request merged**